pkg/pprof/merge_test.go

```diff
@@ -23,10 +24,17 @@ func Test_Merge_Single(t *testing.T) {
 	p, err := OpenFile("testdata/go.cpu.labels.pprof")
 	require.NoError(t, err)
 	var m ProfileMerge
-	require.NoError(t, m.Merge(p.Profile))
+	require.NoError(t, m.Merge(p.Profile.CloneVT()))
+	sortLabels(p.Profile)
 	testhelper.EqualProto(t, p.Profile, m.Profile())
 }
 
+func sortLabels(p *profilev1.Profile) {
+	for _, s := range p.Sample {
+		sort.Sort(LabelsByKeyValue(s.Label))
+	}
+}
+
```

Comment:
Note that `Merge` modifies profiles in place. Therefore, some tests were incorrect. `MergeNoClone` in addition retains the objects; it's quite hard to use the function correctly. I'm almost sure this is causing issues.